tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from azure.mgmt.web.models import (Site, SiteConfig, NameValuePair)
+from c7n_azure.session import Session
+
+from c7n.utils import local_session
+
+
+class FunctionAppUtilities(object):
+    def __init__(self):
+        self.local_session = local_session(Session)
+        self.log = logging.getLogger('custodian.azure.function_app_utils')
+
+    def deploy_webapp(self, app_name, group_name, service_plan, storage_account_name):
+        self.log.info("Deploying Function App %s (%s) in group %s" %
+                      (app_name, service_plan.location, group_name))
+
+        site_config = SiteConfig(app_settings=[])
+        functionapp_def = Site(location=service_plan.location, site_config=site_config)
+
+        functionapp_def.kind = 'functionapp,linux'
+        functionapp_def.server_farm_id = service_plan.id
+
+        site_config.linux_fx_version = 'DOCKER|microsoft/azure-functions-python3.6:latest'
```

Comment:
If we want to move the docker version to a constant class, we can pin the version to: v2.0.11933-alpha
https://hub.docker.com/r/microsoft/azure-functions-python3.6/tags/